### PR TITLE
Update the C++ custom instrumentation example to include resource name

### PR DESCRIPTION
**Pull request by wantsui**
### What does this PR do?

This PR demonstrates how the resource name can be defined for a custom span in C++.

### Motivation

To clarify the current example.

### Preview

https://docs-staging.datadoghq.com/wantsui/resource-name-examples-cpp/tracing/setup_overview/custom_instrumentation/cpp

### Additional Notes

This needs approval from someone on the C++ team.

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Review by dgoffredo — APPROVED** [review]
Looks good to me. I merged in a modification earlier.

File: content/en/tracing/setup_overview/custom_instrumentation/cpp.md [hunk 1]
```diff
@@ -69,15 +69,20 @@ To manually instrument your code, install the tracer as in the [setup examples][
 
 ```cpp
 {
-  // Create a root span.
-  auto root_span = tracer->StartSpan("operation_name");
-  // Create a child span.
+  // Create a root span for the request we're currently processing.
```
Comment (alai97) [inline content/en/tracing/setup_overview/custom_instrumentation/cpp.md:72]:
```suggestion
  // Create a root span for the processing request.
```

File: content/en/tracing/setup_overview/custom_instrumentation/cpp.md [hunk 2]
```diff
@@ -69,15 +69,20 @@ To manually instrument your code, install the tracer as in the [setup examples][
...
+  auto root_span = tracer->StartSpan("get_ingredients");
+  // Set a resource name for the root span.
+  root_span->SetTag(datadog::tags::resource_name, "bologna_sandwich");
+  // Create a child span, with the root span as its parent.
```
Comment (alai97) [inline content/en/tracing/setup_overview/custom_instrumentation/cpp.md:76]:
```suggestion
  // Create a child span with the root span as its parent.
```

File: content/en/tracing/setup_overview/custom_instrumentation/cpp.md [hunk 3]
```diff
@@ -69,15 +69,20 @@ To manually instrument your code, install the tracer as in the [setup examples][
...
+      "cache_lookup",
       {opentracing::ChildOf(&root_span->context())});
-  // Spans can be finished at a specific time ...
+  // Set a resource name for the child span.
+  child_span->SetTag(datadog::tags::resource_name, "ingredients.bologna_sandwich");
+  // Spans can be finished at an explicit time ...
   child_span->Finish();
-} // ... or when they are destructed (root_span finishes here).
+} // ... or otherwise implicitly when their destructor is invoked.
+  // For example, root_span finishes here implicitly.
```
Comment (alai97) [inline content/en/tracing/setup_overview/custom_instrumentation/cpp.md:85]:
```suggestion
} // ... or implicitly when the destructor is invoked.
  // For example, root_span finishes here.
```

**Review by alai97 — CHANGES_REQUESTED** [review]
Looks good pending three copy nits!

**Review by dgoffredo — APPROVED** [review]
still LGTM

File: content/en/tracing/setup_overview/custom_instrumentation/cpp.md (hunk 1 above)
Comment (dgoffredo) [reply content/en/tracing/setup_overview/custom_instrumentation/cpp.md]: Perhaps this is better:

> // Create a root span for the current request.

The verb "processing" is a little awkward and ambiguous here.

Comment (dgoffredo) [reply content/en/tracing/setup_overview/custom_instrumentation/cpp.md]: Ah, my browser was showing me an older version, before this change was merged.

File: content/en/tracing/setup_overview/custom_instrumentation/cpp.md [hunk 4]
```diff
@@ -69,15 +69,20 @@ To manually instrument your code, install the tracer as in the [setup examples][
 
 ```cpp
 {
-  // Create a root span.
-  auto root_span = tracer->StartSpan("operation_name");
-  // Create a child span.
+  // Create a root span for the current request.
```
Comment (wantsui) [inline content/en/tracing/setup_overview/custom_instrumentation/cpp.md:72]: @dgoffredo @alai97 - does this reviewed comment look fine?

**Review by alai97 — APPROVED** [review]

**Pull request merged**